src/org/sosy_lab/java_smt/solvers/princess/PrincessModel.java

```diff
@@ -240,14 +253,76 @@ public String toString() {
   @Override
   public void close() {}
 
+  /** Tries to determine the Sort of a Term. */
+  private Sort getSort(IExpression pTerm) {
+    // Just using sortof() won't work as Princess may have simplified the original term
+    // FIXME: This may also affect other parts of the code that use sortof()
+    if (pTerm instanceof ITimes) {
+      ITimes times = (ITimes) pTerm;
+      return getSort(times.subterm());
+    } else if (pTerm instanceof IPlus) {
+      IPlus plus = (IPlus) pTerm;
+      return getSort(plus.apply(0));
+    } else if (pTerm instanceof IFormula) {
+      return creator.getBoolType();
+    } else {
+      // TODO: Do we need more cases?
```

Comment:
It would be good to look into this before merging.

Reply:
can be done after merging